cpp/oneapi/dal/io/detail/load_graph.hpp

```diff
@@ -53,126 +54,127 @@ edge_list<std::int32_t> load_edge_list(const std::string &name) {
 
 template <typename Graph>
 void convert_to_csr_impl(const edge_list<vertex_type<Graph>> &edges, Graph &g) {
-    auto layout = oneapi::dal::preview::detail::get_impl(g);
-    using vertex_t = typename Graph::vertex_type;
-    using vector_vertex_t = typename Graph::vertex_set;
-    using vector_edge_t = typename Graph::edge_set;
-    using allocator_t = typename Graph::allocator_type;
-    using vertex_size_t = typename Graph::vertex_size_type;
-
     if (edges.size() == 0) {
-        layout->_vertex_count = 0;
-        layout->_edge_count = 0;
         throw invalid_argument("Empty edge list");
     }
 
-    vertex_t max_node_id = edges[0].first;
+    using vertex_t = typename Graph::vertex_type;
+    using edge_t = typename Graph::edge_type;
+    using vector_vertex_t = typename Graph::vertex_set;
+    using atomic_t = typename daal::services::Atomic<vertex_t>;
+
+    vertex_t max_id = edges[0].first;
     for (auto u : edges) {
-        vertex_t max_id_in_edge = std::max(u.first, u.second);
-        max_node_id = std::max(max_node_id, max_id_in_edge);
+        vertex_t edge_max = std::max(u.first, u.second);
+        max_id = std::max(max_id, edge_max);
     }
 
-    vertex_t _unf_vertex_count = max_node_id + 1;
+    const vertex_t n_vertex = max_id + 1;
 
-    using atomic_t = typename daal::services::Atomic<vertex_t>;
-    using allocator_atomic_t =
-        typename std::allocator_traits<allocator_t>::template rebind_alloc<atomic_t>;
+    auto layout = oneapi::dal::preview::detail::get_impl(g);
+    auto &allocator = layout->_allocator;
+    layout->_vertex_count = n_vertex;
+
+    void *degrees_vec_void =
+        (void *)allocator.allocate(n_vertex * (sizeof(atomic_t) / sizeof(char)));
```

Comment:
1. Please use C++ conversions instead of C-style conversions
2. `char` -> `std::int8_t` or `std::uint8_t`

Reply:
Why you use such strange expression for allocation: `(sizeof(atomic_t) / sizeof(char))` ?
Method `allocate()` for standard allocators returns the pointer to exact type `T` what you've specified. Why not use this type here?

Reply:
allocator for graphs returns char type due to proposal